src/merge.js

```
@@ -6,14 +6,24 @@ const { inspect, isDeepStrictEqual } = require('util')
 // is used, it is possible to merge `_redirects` to `netlify.toml` by prepending
 // its rules to `netlify.toml` `redirects` field.
 const mergeRedirects = function ({ fileRedirects = [], configRedirects = [] }) {
-  validateArray(fileRedirects)
-  validateArray(configRedirects)
-  return [...fileRedirects, ...configRedirects].filter(isUniqueRedirect)
+  const errors = validateArrays(fileRedirects, configRedirects)
+  if (errors.length !== 0) {
+    return { redirects: [], errors }
+  }
```

If we want to return the errors while keeping the redirects result shouldn't we still return one of the arrays even if the other turns out to be invalid?

---

Good point! Fixed.
I also fixed `netlify-headers-parser` in https://github.com/netlify/netlify-headers-parser/pull/12